tests/components/tradfri/test_switch.py

```diff
@@ -1,160 +1,102 @@
+from __future__ import annotations
 
-from unittest.mock import MagicMock, Mock, PropertyMock, patch
+import json
+from typing import Any
+from unittest.mock import MagicMock, Mock
 
 import pytest
+from pytradfri.const import ATTR_REACHABLE_STATE
 from pytradfri.device import Device
 from pytradfri.device.socket import Socket
-from pytradfri.device.socket_control import SocketControl
-
-from .common import setup_integration
-
-
-@pytest.fixture(autouse=True, scope="module")
-def setup(request):
-    """Set up patches for pytradfri methods."""
-    with patch(
-        "pytradfri.device.SocketControl.raw",
-        new_callable=PropertyMock,
-        return_value=[{"mock": "mock"}],
-    ), patch(
-        "pytradfri.device.SocketControl.sockets",
-    ):
-        yield
-
-
-def mock_switch(test_features=None, test_state=None, device_number=0):
-    """Mock a tradfri switch/socket."""
-    if test_features is None:
-        test_features = {}
-    if test_state is None:
-        test_state = {}
-    mock_switch_data = Mock(**test_state)
-
-    dev_info_mock = MagicMock()
-    dev_info_mock.manufacturer = "manufacturer"
-    dev_info_mock.model_number = "model"
-    dev_info_mock.firmware_version = "1.2.3"
-    _mock_switch = Mock(
-        id=f"mock-switch-id-{device_number}",
-        reachable=True,
-        observe=Mock(),
-        device_info=dev_info_mock,
-        has_light_control=False,
-        has_socket_control=True,
-        has_blind_control=False,
-        has_signal_repeater_control=False,
-        has_air_purifier_control=False,
-    )
-    _mock_switch.name = f"tradfri_switch_{device_number}"
-    socket_control = SocketControl(_mock_switch)
 
-    # Store the initial state.
-    setattr(socket_control, "sockets", [mock_switch_data])
-    _mock_switch.socket_control = socket_control
-    return _mock_switch
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.components.tradfri.const import DOMAIN
+from homeassistant.const import STATE_OFF, STATE_ON, STATE_UNAVAILABLE
+from homeassistant.core import HomeAssistant
 
+from .common import setup_integration, trigger_observe_callback
 
-async def test_switch(hass, mock_gateway, mock_api_factory):
-    """Test that switches are correctly added."""
-    state = {
-        "state": True,
-    }
+from tests.common import load_fixture
 
-    mock_gateway.mock_devices.append(mock_switch(test_state=state))
-    await setup_integration(hass)
 
-    switch_1 = hass.states.get("switch.tradfri_switch_0")
-    assert switch_1 is not None
-    assert switch_1.state == "on"
+@pytest.fixture(scope="module")
+def outlet() -> dict[str, Any]:
+    """Return an outlet response."""
+    return json.loads(load_fixture("outlet.json", DOMAIN))
 
 
-async def test_switch_observed(hass, mock_gateway, mock_api_factory):
-    """Test that switches are correctly observed."""
-    state = {
-        "state": True,
-    }
+@pytest.fixture
+def socket(outlet) -> Socket:
```

Comment:
For the sake of consistency with the other functions in this file:

```suggestion
def socket(outlet: dict[str, Any]) -> Socket:
```

Reply:
Thanks!